packages/atti-components/src/components/Slider/index.js

```diff
@@ -0,0 +1,104 @@
+import React, {useContext, useEffect, useState} from 'react'
+import PropTypes from 'prop-types'
+import styled, {ThemeContext} from 'styled-components';
+import {map, max, min, minBy, round} from 'lodash';
+import SmallText from "../Typographies/SmallText";
+import SliderRail from "./styles/SliderRail";
+import SliderIndicator from "./styles/SliderIndicator";
+import SliderValue from "./styles/SliderValue";
+import SliderStep from "./styles/SliderStep";
+
+const Slider = ({isStepRestricted, isValueDisplayed, onValueChange, precision, range, steps}) => {
+  const {components: {slider: {sliderRail}}} = useContext(ThemeContext)
+  const [isPressed, setIsPressed] = useState(false)
+  const [indicatorPos, setIndicatorPos] = useState(0)
+  const [xMousePosition, setXMousePosition] = useState(0)
+
+  const pxSliderSize = sliderRail.sizes.width.slice(0, -2)
+  // Add event regarding the indicator motion, remove it when Slider is unmounted
+  useEffect(() => {
```

Comment:
This useEffect is missing the dependencies list as 2nd arg

Reply:
In order to reproduce the ComponentDidMount and ComponentWillUnmount behavior that is wanted here, the useEffect Hook need to have no second arg or an empty array [].

Reply:
`useEffect(function)` & `useEffect(function, [])` are completely different.

The first one (no second arg) will run at **EACH** render. The second one (empty array) will only render after your component has mounted for the first time.

You can check out theirs docs https://fr.reactjs.org/docs/hooks-reference.html#useeffect for more info :)

Reply:
Ok, in order to optimize the event handling I finally decided to run this side effect only on isPressed state update.